Packs/SophosCentral/Integrations/SophosCentral/SophosCentral.py

```diff
@@ -0,0 +1,1764 @@
+
+
+def fetch_incidents(client: Client, last_run: Dict[str, int],
+                    first_fetch_time: str, fetch_severity: Optional[List[str]],
+                    fetch_category: Optional[List[str]],
+                    max_fetch: Optional[str]) -> Tuple[Dict[str, int], List[dict]]:
+    """
+    Fetch incidents (alerts) each minute (by default).
+    Args:
+        client (Client): Sophos Central Client.
+        last_run (dict): Dict with last_fetch object,
+                                  saving the last fetch time(in millisecond timestamp).
+        first_fetch_time (dict): Dict with first fetch time in str (ex: 3 days ago).
+        fetch_severity (list(str)): Severity to fetch.
+        fetch_category (list(str)): Category(s) to fetch.
+        max_fetch (str): Max number of alerts to fetch.
+    Returns:
+        Tuple of next_run (millisecond timestamp) and the incidents list
+    """
+    last_fetch_timestamp = last_run.get('last_fetch', None)
+    fetch_category = None if not fetch_category else fetch_category
+    fetch_severity = None if not fetch_severity else fetch_severity
+    limit = None if not max_fetch else int(max_fetch)
+
+    if last_fetch_timestamp:
+        last_fetch_date = datetime.fromtimestamp(last_fetch_timestamp / 1000)
+        last_fetch = last_fetch_date
+    else:
+        first_fetch_date = dateparser.parse(first_fetch_time)
+        last_fetch = first_fetch_date
+
+    incidents = []
+    next_run = last_fetch
+    alerts = client.search_alert(last_fetch.strftime(DATE_FORMAT)[:-4] + 'Z', None, None,
+                                 fetch_category, None, fetch_severity, None, limit)
+    data_fields = ['id', 'description', 'severity', 'raisedAt', 'allowedActions', 'managedAgentId',
+                   'category', 'type']
+    for alert in alerts.get('items', []):
+        alert = create_alert_output(alert, data_fields)
+        alert_created_time = alert.get('raisedAt')
+        alert_id = alert.get('id')
+        incident = {
+            'name': f'Sophos Central Alert {alert_id}',
+            'occurred': alert_created_time,
+            'rawJSON': json.dumps(alert)
+        }
+        incidents.append(incident)
+        created_time = datetime.strptime(alert_created_time, DATE_FORMAT)
+        if created_time > next_run:
+            next_run = created_time
+
+    next_run = next_run + timedelta(milliseconds=1)
+    next_run_timestamp = int(datetime.timestamp(next_run) * 1000)
+    return {'last_fetch': next_run_timestamp}, incidents
+
+
+def test_module(client: Client) -> str:
+    """
+    Test the validity of the connection and the
+
+    Args:
+        client (Client): Sophos Central client
+
+    Returns:
+        'ok' if test passed, anything else will fail the test.
+    """
+    try:
+        result = client.list_alert(1)
+        if result and result.get('pages'):
+            return 'ok'
+        return str(result)
```

Comment:
I don't know but it might be possible for the API to return an error that isn't initally caught. Anything other than the wanted API response (200 with data inside) is considered an error.

Reply:
@Lior-Dahan-qm If you will only expects a 200, the baseclient `_http_request` will raise an error otherwise.
i'm saying that i'm not sure that you need to check if the result/response exist each time.
but if you do - handle the else case appropriately.

Reply:
I haven't encountered any cases where that was the case, so I can't say I know of any appropriate cases to check for.